File: platform/Leetcode/src/UglyNumber.java
import java.util.Scanner;

public class UglyNumber {
    public static void main (String[] args) {
        System.out.print("Enter a positive integer: ");
        Scanner sc = new Scanner(System.in);
        int number = Integer.valueOf(sc.nextLine());
        boolean result = isUgly(number);
        
        if (result) {
            System.out.println("The number is ugly");
        } else {
            System.out.println("The number is NOT ugly");
        }
        
    }
    
    public static boolean isUgly(int number){
        if (number <= 0) return false;
        if (number == 1) return true;    
        if (number % 2 ==0) return isUgly(number/2);
        if (number % 3 ==0) return isUgly(number/3);
        if (number % 5 ==0) return isUgly(number/5);
        return false;
    }
}

File: platform/Leetcode/src/ugly_number.py
# Use recursion
def is_ugly(number):
    ''' Function returns True/False when a positive integer is determined to be an ugly number.
    Ugly numbers are positive numbers whose prime factors only include 2, 3, 5.
    The number 1 is considered as ugly. 

    '''
    
    if (number <= 0):
        return False
    elif (number == 1):
        return True    
    elif (number % 2 == 0):
        return is_ugly(number / 2)
    elif (number % 3 == 0):
        return is_ugly(number / 3)
    elif (number % 5 == 0):
        return is_ugly(number / 5)
    return False

if __name__ == '__main__':
    number = input("Enter a positive integer value: ")
    result = is_ugly(int(number))
    answer = '' if result else ' NOT '    
    print('This integer is %s ugly.' %(answer))